smartfile/__init__.py

```diff
@@ -124,8 +124,40 @@ def put(self, endpoint, id=None, **kwargs):
     def post(self, endpoint, id=None, **kwargs):
         return self._request('post', endpoint, id=id, data=kwargs)
 
-    def delete(self, endpoint, id=None, **kwargs):
-        return self._request('delete', endpoint, id=id, data=kwargs)
+    def delete(self, deletefile):
+        try:
+            return self.post('/path/oper/remove', path=deletefile)
+        except KeyError:
+            raise Exception("Destination file does not exist")
+        
+    def upload(self, usrfile):
+        # needed to split the tuple
+        newtuple = usrfile
+        # splits tuple to allow ability to remove "/" at the end, if present 
+        ourfile, newdata = newtuple
+        if ourfile[-1:] == "/":
+             raise Exception("Can only upload files")
+        return self.post('/path/data/', file=usrfile)  
+        
+    def download(self, downloadfile):
+        return self.get('/path/data/', downloadfile)
+    
+    def move(self, sourcefile, destination):
+        # check destination folder for / at end
+        if destination[-1:] != "/":
+            raise Exception("Destination folder requires a / at end")
+        # check destination folder for / at begining
+        if destination[:-1] != "/":
+            destination = "/" + destination
+        try:
+            t = self.post('/path/oper/move/', src=sourcefile, dst=destination)
+        except KeyError:
+            raise Exception("Destination directory does not exist")
+        while True:
+            s = self.get('/task', t['uuid'])
+            if s['result']['status'] == 'SUCCESS':
+                break
```

Comment:
The speedup in is this case negligible. We're not in a loop or a critical section. Most of the times this will be a one timer, with a far worse bottleneck on top of it which is represented by the network (it's a client). Better to prefer readability over performance (in this case and in general, most of the times). E.g. at a first glance I wasn't sure whether `[:-1]` returned the last or the first part of the string so I had to open a terminal and figure it out myself, hence the call for readability over performance.